### PR TITLE
Add Billing Account resource to Billing RP.

### PR DESCRIPTION
**Pull request by wilcobmsft**
**Background**: Billing Account is a new ARM resource we're introducing, and it is a projection of state mostly mastered in commerce. This resource can be used by developers to assign subscription creation permissions to other users or service principals.

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [X] The title of the PR is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [X] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [X] If applicable, the PR references the bug/issue that it fixes.
- [X] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [X] My spec meets the review criteria:
  - [X] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [X] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [X] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-03-01-preview/examples/InvoicesExpand.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{
+  "parameters": {
+    "api-version": "2018-03-01-preview",
+    "subscriptionId": "subid",
+    "$top": "1",
```
Comment (jhendrixMSFT) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2018-03-01-preview/examples/InvoicesExpand.json]: This type is mismatched, per spec it should be an int not a string, and is causing model validation [failure](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/344048908).

**Review by jhendrixMSFT — CHANGES_REQUESTED** [review]

**Comment by jhendrixMSFT** [comment]
@wilcobmsft Has this API been reviewed by the ARM team yet?

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-03-01-preview/examples/BillingAccountsGet.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+  "parameters": {
```
Comment (jhendrixMSFT) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2018-03-01-preview/examples/BillingAccountsGet.json]: Required parameter "get" is missing, see the [log](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/344412418).

Comment (wilcobmsft) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2018-03-01-preview/examples/BillingAccountsGet.json]: Fixed.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-03-01-preview/examples/InvoicesExpand.json (hunk 1 above)
Comment (wilcobmsft) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2018-03-01-preview/examples/InvoicesExpand.json]: Fixed

**Comment by wilcobmsft** [comment]
@jhendrixMSFT: Not yet - I've reached out to them with a link to this pr and a description of our overall plan. This PR should not be merged until someone from ARM has provided sign-off.

**Comment by jhendrixMSFT** [comment]
Ping @ravbhatnagar

**Comment by ravbhatnagar** [comment]
Looks good. Reviewed in person.

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**